src/main/java/org/broadinstitute/hellbender/utils/dragstr/DragstrParams.java

```diff
@@ -0,0 +1,478 @@
+package org.broadinstitute.hellbender.utils.dragstr;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.tools.dragstr.DragstrHyperParameters;
+import org.broadinstitute.hellbender.tools.walkers.genotyper.afcalc.AlleleFrequencyCalculator;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.IntStream;
+
+/**
+ * Holds the values of the DRAGstr model parameters for different combinations of repeat unit length (period) and
+ * number of repeat units.
+ * <p>
+ *     This class is immutable.
+ * </p>
+ * <h3>Parameter matrices</h3>
+ * <p>
+ *     The DRAGstr model are composed of three matrices with the same dimensions (period x repeat-length):
+ *     <dl>
+ *         <dt>GOP</dt>
+ *         <dd>gap-opening-penalty table where we store the Phred scale score that we use for pairhmm match to insert or
+ *         delete transitions</dd>.
+ *         <dt>GCP</dt>
+ *         <dd>gap-continuation-penalty table where we store the Phred scale score that we use for pair-hmm insert to insert, and
+ *             deletion to deletion
+ *             </dd>.
+ *         <dt>API</dt>
+ *         <dd>adjusted prior probability of an indel occurring at a position in the reference with an STR.</dd>
+ *     </dl>
+ * </p>
+ * <h3>Text file format</h3>
+ * If P is the maximum period length and L is the maximum repeat-length:
+ * <p>
+ *     <pre>
+ *         ############################
+ *         # DragstrParams
+ *         # -------------------------
+ *         # annotation1 = value1
+ *         # annotation2 = value2
+ *         # ...
+ *         ###########################
+ *              1      2     3     4 ...     L
+ *         GOP:
+ *          gop11  gop12 gop13 gop14 ... gop1L
+ *          gop21  gop22 ...
+ *          ...
+ *          gopP1  gopP2 ...             gopPL
+ *         GCP:
+ *          gcp11  gcp12 gcp13 ...
+ *          ...
+ *          gcpP1  apiP2 ...     ...     apiPL
+ *         API:
+ *          api11  api12 ...
+ *          ..
+ *          apiP1  apiP2 ...             apiPL
+ *     </pre>
+ * </p>
+ * <p>
+ *     Currently there is not a fix/standard set of annotations but is just a mechanism to add some additional
+ *     metadata that might be useful.
+ * </p>
+ */
+public final class DragstrParams {
```

Comment:
This is a stylistic comment but this class is essentially a codec that handles both writing and reading of these objects from a table in that is just 3 2d arrays written out. That is somewhat obfuscated by the structure of this class. I would advocate pulling out a `dragstrTableCodec` that this classes constructor/writer both call out to.

Reply:
I need you to be more specific.... I can see the existence of a class TableCodec but it forces to  indicate a genome:loc for each row.... The information in dragstr params is not locatable to any genome location. Perhaps you are talking about another "codec" framework (TableReader, TableWriter?) or pherhaps you suggest to use a dummy location (E.g. chr1:1) everywhere?

Beside that, I think I commented about this when I worked on my review but perhaps the comment was lost or you missed it due to the way I posted it (was within my own review comments).... 

So it seem that here you are suggesting (although you don't say it explicitly) to "melt" those three tables into one where there are columns for the cell coordinates (period, repeat length) and 3 column for the values of the original tables. While that make it easier to parse by the computer as it is a more standard format, it kinda makes it loose human readability where you can at a glance see how period and repeat length affect the different parameters; if you peak into one of those files you will see what I mean with this immediately. 

Since the code is already in place and the work is done I suggest that we keep it the format as it is based on this advantage.

Reply:
I did not mean to invoke the actual `TableCodec` I apologize. What I mean is perhaps we can separate out a "codec" class that handles reading/writing the dragstr table that is kept separate from the list. This class is mostly various writer methods anyway and it makes it somewhat difficult to find what methods apply to the class and what are just for various steps of formatting before the end. This is merely a stylistic suggestion.

Reply:
Ok. I will exrtract the readers and writer classes from inners to top-level.